tests/devapps/Xamarin Manual/Xamarin Manual/Xamarin Manual/MainPage.xaml.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace Xamarin_Manual
+{
+
+    // Learn more about making custom code visible in the Xamarin.Forms previewer
+    // by visiting https://aka.ms/xamarinforms-previewer
+    [DesignTimeVisible(false)]
+    public partial class MainPage : ContentPage
+    {
+        private IPublicClientApplication _pca;
+        public static object CurrentActivity { get; set; }
+
+        /// <summary>
+        /// The ClientID is the Application ID found in the portal (https://go.microsoft.com/fwlink/?linkid=2083908). 
+        /// You can use the below id however if you create an app of your own you should replace the value here.
+        /// </summary>
+        
+
+        public MainPage()
+        {
+            InitializeComponent();
+            pckWebView.SelectedIndex = 0;
+
+            CreatePca();
+        }
+
+        private async void atsBtn_Clicked(object sender, EventArgs ea)
+        {
+            try
+            {
+                var accounts = await _pca.GetAccountsAsync().ConfigureAwait(false);
+                var result = await _pca.AcquireTokenSilent(AuthConfig.Scopes, accounts.FirstOrDefault()).ExecuteAsync().ConfigureAwait(false);
+                UpdateStatus($"ATS Token! Valid for: {(result.ExpiresOn - DateTimeOffset.Now).TotalMinutes} min");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                UpdateStatus("Failed! " + e);
```

Comment:
maybe mentioned which btn was hit to generate the exception in the error message?